pynio/rest.py

```diff
@@ -12,14 +17,25 @@ def __init__(self, host='127.0.0.1', port=8181, creds=None):
         self._creds = creds or ('User', 'User')
         self._url = 'http://{}:{}/{}'.format(host, port, '{}')
 
-    def _get(self, endpoint, timeout=None, data=None):
+    def _get(self, endpoint, timeout=None, data=None, retry=0):
+        '''Performs a get with some amounts of retrys'''
         if data is not None:
             data = json.dumps(data)
-        r = requests.get(self._url.format(endpoint),
-                         auth=self._creds,
-                         timeout=timeout,
-                         data=data)
-        r.raise_for_status()
+        for i in range(retry + 1):
+            try:
+                r = requests.get(self._url.format(endpoint),
+                                 auth=self._creds,
+                                 timeout=timeout,
+                                 data=data)
+                r.raise_for_status()
+                break
+            except requests.exceptions.ConnectionError as e:
+                log.warning("Failure connecting in _get")
+                E = e
```

Comment:
wait no, you are wrong. If it raised a different error it would raise it out of the loop altogether (it would be raised properly). I pushed a unit test for just that